core/state_transition.go

```diff
@@ -535,9 +535,9 @@ func (st *StateTransition) innerTransitionDb() (*ExecutionResult, error) {
 
 	// Check that we are post bedrock to enable op-geth to be able to create pseudo pre-bedrock blocks (these are pre-bedrock, but don't follow l2 geth rules)
 	// Note optimismConfig will not be nil if rules.IsOptimismBedrock is true
-	if optimismConfig := st.evm.ChainConfig().Optimism; optimismConfig != nil && rules.IsOptimismBedrock {
+	if optimismConfig := st.evm.ChainConfig().Optimism; optimismConfig != nil && rules.IsOptimismBedrock && !st.msg.IsDepositTx {
```

Comment:
Why is this additional restriction added?

Reply:
L1CostFunc used to accept "IsDepostTx" to skip the statedb lookup internally, but I thought it was cleaner to just clarify not to call the function on deposit txs at all.

Reply:
obscured because github doesn't handle diffs well when files are renamed, but the old cost func signature was:

return func(blockNum uint64, blockTime uint64, dataGas RollupGasData, isDepositTx bool)